test/llvm/parallel_loop_access/different_numbers.chpl

```diff
@@ -0,0 +1,53 @@
+proc start_loop1() {}
+proc end_loop1() {}
+
+proc start_loop2() {}
+proc end_loop2() {}
+
+proc start_loop3() {}
+proc end_loop3() {}
+
+//Check whether we generate different metadata number for loops
+proc loop (A, B, n) {
+  for i in vectorizeOnly(1..n) {
+    //CHECK-LABEL: start_loop1
+    start_loop1();
+
+    //CHECK: llvm.mem.parallel_loop_access ![[LOOP1:[0-9]+]]
+    A[i] = 3*B[i];
+
+    //CHECK-LABEL: end_loop1
+    end_loop1();
+  }
+
+  for i in vectorizeOnly(1..n) {
+    //CHECK-LABEL: loop2
+    start_loop2();
+
+    //CHECK: llvm.mem.parallel_loop_access ![[LOOP2:[0-9]+]]
+    //CHECK-NOT: llvm.mem.parallel_loop_access ![[LOOP1]]
+    A[i] = 5*B[i];
+    for j in vectorizeOnly(1..n) {
+      //CHECK-LABEL: start_loop3
+      start_loop3();
+
+      //CHECK: llvm.mem.parallel_loop_access ![[LOOP3:[0-9]+]]
+      //CHECK-NOT: llvm.mem.parallel_loop_access ![[LOOP2]]
```

Comment:
Can you point me to some LLVM documentation explaining what you mean by `group parallel` ? As I understand `llvm.mem.parallel_loop_access`, it indicates:

> no loop carried memory dependence exist between it and other instructions denoted with the same loop identifier

In your example, the inner loop and the outer loop both have no loop-carried dependencies (that is, it would be a user error if the program depended upon the loop iteration order). There *is* a dependency between the first statement in the outer loop and the inner loop. If I understand correctly, my Allen & Kennedy textbook calls such a thing a *loop-independent dependency*.